compiler/src/iree/compiler/DispatchCreation/SetEncoding.cpp

```diff
@@ -310,28 +311,17 @@ struct FoldFillWithSetEncoding final
 // Set padding encodings
 //===---------------------------------------------------------------------===//
 
-struct PaddedValue {
-  Value paddedValue;
-  SmallVector<Value> dynamicDims;
-};
-
-// For a given `operand`, if its producer is a `flow.dispatch.region`,
-// generate a new value for `operand` that has the padding encoding.
-// The producer `flow.dispatch.region` need not be the immediate defining op
-// of `operand`. This method tracks through operations like
-// `tensor.expand_shape/tensor.collapse_shape` to get to the producer dispatch.
-// Once the producer dispatch is found, its result is modified to be of the same
-// type as `operand` but with the padding encodings. To keep things consistent,
-// the operations that are encountered before getting to the original producing
-// `flow.dispatch.region` are replicated into the producer dispatch.
-static std::optional<PaddedValue> padProducerOfValue(RewriterBase &rewriter,
-                                                     Value operand) {
+// Utility to return the producer dispatch region op and op result. 'opChain' is
+// an output parameter and allows returning the chain of operations being looked
+// past during the traversal to find the producer dispatch.
+static std::optional<std::pair<IREE::Flow::DispatchRegionOp, OpResult>>
+getProducerDispatchAndValue(Value operand,
+                            SmallVectorImpl<Operation *> &opChain) {
```

Comment:
I'd expect the function to return `std::optional<std::pair<OpResult, SmallVector<Operation*>>>`, because `OpResult` has enough information. You can always cast the owner to `DispatchRegionOp`.

You can do something like `std::tie(dispatchOpResult, std::ignore) = maybeProducerDispatchAndValue.value()` when you don't care about the operation chain. Then you don't need the other wrapper.

Reply:
Thanks, good suggestion, updated it.